src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -0,0 +1,205 @@
+                                                           const CharSpan & serialNumberHexStr)
+{
+    std::ifstream file(mDeviceAttestationRevocationSetPath);
+    if (!file.is_open())
+    {
+        ChipLogError(NotSpecified, "Failed to open file: %s", mDeviceAttestationRevocationSetPath);
+        return false;
+    }
+
+    // Parse the JSON data incrementally
+    Json::CharReaderBuilder readerBuilder;
+    Json::Value jsonData;
+    std::string errs;
+
+    bool parsingSuccessful = Json::parseFromStream(readerBuilder, file, &jsonData, &errs);
+
+    // Close the file as it's no longer needed
+    file.close();
+
+    if (!parsingSuccessful)
+    {
+        ChipLogError(NotSpecified, "Failed to parse JSON: %s", errs.c_str());
+        return false;
+    }
+
+    std::string issuerName   = std::string(issuerNameBase64Str.data(), issuerNameBase64Str.size());
+    std::string serialNumber = std::string(serialNumberHexStr.data(), serialNumberHexStr.size());
+    std::string akid         = std::string(akidHexStr.data(), akidHexStr.size());
+
+    for (const auto & revokedSet : jsonData)
+    {
+        if (revokedSet["issuer_name"].asString() != issuerName)
+        {
+            continue;
+        }
+        if (revokedSet["issuer_subject_key_id"].asString() != akid)
+        {
+            continue;
+        }
+        for (const auto & revokedSerialNumber : revokedSet["revoked_serial_numbers"])
+        {
+            if (revokedSerialNumber.asString() == serialNumber)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetAKIDHexStr(const ByteSpan & certDer, MutableCharSpan & outAKIDHexStr)
+{
+    uint8_t akidBuf[kAuthorityKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+
+    ReturnErrorOnFailure(ExtractAKIDFromX509Cert(certDer, akid));
+
+    return BytesToHexStr(akid, outAKIDHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetSerialNumberHexStr(const ByteSpan & certDer, MutableCharSpan & outSerialNumberHexStr)
+{
+    uint8_t serialNumberBuf[kMaxCertificateSerialNumberLength] = { 0 };
+    MutableByteSpan serialNumber(serialNumberBuf);
+
+    ReturnErrorOnFailure(ExtractSerialNumberFromX509Cert(certDer, serialNumber));
+    return BytesToHexStr(serialNumber, outSerialNumberHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetIssuerNameBase64Str(const ByteSpan & certDer,
+                                                                 MutableCharSpan & outIssuerNameBase64String)
+{
+    uint8_t issuerBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
+    MutableByteSpan issuer(issuerBuf);
+
+    ReturnErrorOnFailure(ExtractIssuerFromX509Cert(certDer, issuer));
+    VerifyOrReturnError(outIssuerNameBase64String.size() >= BASE64_ENCODED_LEN(issuer.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint32_t encodedLen = Base64Encode32(issuer.data(), static_cast<uint32_t>(issuer.size()), outIssuerNameBase64String.data());
```

Comment:
This can be simply `Base64Encode`.  I don't know what the `32` suffix is about.

Reply:
32 suffix returns the 32 bit value. `Base64Encode` returns 16 bit value. Using `Base64Encode`.